### PR TITLE
update vars and versions.tf

### PR DESCRIPTION
**Pull request by marwinbaumannsbp**
- add support for setting agent_pool_id, execution_mode, ssh_key_id, clear_text_hcl_variables,  sensitive_hcl_variables
- make module more generic, change github mentions to repository
- update versions.tf

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -71,3 +74,24 @@ resource "tfe_variable" "sensitive_terraform_variables" {
...
+resource "tfe_variable" "sensitive_hcl_variables" {
+  for_each = var.sensitive_hcl_variables
+
+  key          = each.key
+  value        = each.value.sensitive
+  category     = "terraform"
+  hcl          = true
+  sensitive    = true
+  workspace_id = tfe_workspace.default.id
+}
```
Comment (svanharmelen) [inline main.tf:97]: Please place these two resources between the `env` and `terraform` ones and make sure to end the file with a newline.

File: versions.tf [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
-  required_version = ">= 0.12.0"
-}
+  required_version = ">= 0.13.0"
+
+  required_providers {
+    tfe = {
+      source = "hashicorp/tfe"
+    }
+  }
+}
```
Comment (svanharmelen) [inline versions.tf:9]: Please make sure to end the file with a newline. Probably best to configure your editor to automatically insert a newline at the end of the file.

File: README.md [hunk 3]
```diff
@@ -16,29 +22,36 @@ Terraform run will fail.
 ## Inputs
 
 | Name | Description | Type | Default | Required |
-|------|-------------|------|---------|:-----:|
+|------|-------------|------|---------|:--------:|
+| name | A name for the Terraform workspace | `string` | n/a | yes |
+| repository\_name | The GitHub or GitLab repository to connect the workspace to | `string` | n/a | yes |
+| repository\_owner | The GitHub organization or GitLab namespace that owns the repository | `string` | n/a | yes |
+| terraform\_organization | The Terraform Enterprise organization to create the workspace in | `string` | n/a | yes |
```
Comment (svanharmelen) [inline README.md:29]: Some of these seem out of order...

**Review by svanharmelen — CHANGES_REQUESTED** [review]
Looks good, just a few minor comments.

File: README.md (hunk 3 above)
Comment (marwinbaumannsbp) [reply README.md:29]: Yes in most repositories we create the readme with this flag: --sort-inputs-by-required, and I like this its still alphabetically ordered but it puts the required vars on top. Changed the github actions in this repo to match the github actions we use for the other mcaf repos

**Review by svanharmelen — APPROVED** [review]

**Review by svanharmelen — DISMISSED** [review]
Hold on... Can we keep it to one repo var?

**Review by stefanwb — APPROVED** [review]

**Pull request merged**